compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/Scrollable.kt

```diff
@@ -458,13 +454,11 @@ private suspend fun ScrollingLogic.animatedDispatchScroll(
 private fun Modifier.mouseWheelInput(
     key1: Any?,
     key2: Any?,
-    onMouseWheel: suspend PointerInputScope.(PointerEvent) -> Boolean
+    onMouseWheel: PointerInputScope.(PointerEvent) -> Boolean
 ) = pointerInput(key1, key2) {
-    coroutineScope {
-        while (isActive) {
-            val event = awaitPointerEventScope {
-                awaitScrollEvent()
-            }
+    awaitPointerEventScope {
+        while (true) {
```

Comment:
Just following the example of some other implementations (like https://github.com/JetBrains/compose-multiplatform-core/blob/32de9dd24cb6eff254b2147d4c8d83e5c2430959/compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/ForEachGesture.kt#L100), we can do the following:

```suggestion
    val currentContext = currentCoroutineContext()
    awaitPointerEventScope {
        while (currentContext.isActive) {
```

Otherwise, as I understand, the coroutine can hang. It will never receive a continuation anyway, so it probably remains unnoticed. Although, I'm not sure how to test such a behaviour correctly.

Reply:
Discussed it with @igordmn some time ago. As far as I remember, `awaitPointerEvent` suspends the execution and coroutine can be terminated without explicit checks in this case.